### PR TITLE
Delete k8s storage classes on model destroy

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Some small fixes.
1. cut and paste storage code from state/addmachine.go to state/storage.go
2. makefile driveby to ensure make build is less verbose
3. when deleting a k8s model, remove any storage classes created for the model

## QA steps

deploy a k8s charm with storage
destroy model, check storage class is gone

### CODE REVIEW

File: caas/kubernetes/provider/k8s.go [hunk 1]
```diff
@@ -126,6 +126,18 @@ func (*kubernetesClient) Provider() caas.ContainerEnvironProvider {
...
+	modelSelector := fmt.Sprintf("%s==%s", labelModel, k.namespace)
+	err := k.StorageV1().StorageClasses().DeleteCollection(&v1.DeleteOptions{
+		PropagationPolicy: &defaultPropagationPolicy,
+	}, v1.ListOptions{
+		LabelSelector: modelSelector,
+	})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return errors.Annotate(err, "deleting model storage classes")
+	}
+
```
Comment (ycliuhw) [inline caas/kubernetes/provider/k8s.go]: wondering if it will break or not if we delete storageClass before deleting NS.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
[move common storage funcs out of addmachine](https://trello.com/c/BiNXVD9R/429-move-common-storage-funcs-out-of-addmachine)

**Comment by wallyworld** [comment]
[deleting a k8s model, also delete storage classes](https://trello.com/c/r8HBEj3V/421-deleting-a-k8s-model-also-delete-storage-classes)

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**